zulipterminal/api_types.py

```diff
@@ -503,6 +503,12 @@ class RealmUserEvent(TypedDict):
     person: RealmUserEventPerson
 
 
+class SubmessageEvent(TypedDict):
+    type: Literal["submessage"]
+
+    message_id: int
+
```

Comment:
Oops, got it. Added the other fields for SubmessageEvent, is this fine?

Reply:
For `SubmessageEvent` this is fine, though we might drop `id` since I believe that's an event identifier which is common to all events.

Re the typing more generally, I was mainly looking to see if we could improve the submessage(s) typing in the `Message` typedict - I did mention that above. Combined with the definition of SubmessageEvent, then it would be clearer how compatible these two types are, including via mypy.

Reading again, some of my comments above may not be applicable, since the content is a string.